warp-core/src/main/scala/com/workday/warp/utils/AnnotationReader.scala

```diff
@@ -21,158 +17,63 @@ import scala.util.Try
-        Logger.warn(s"detected overloaded methods for signature $testId, annotation processing may not work as expected.")
-      }
-      methods.headOption
-    }
-  }
-
-
-
-  /**
-   * Reads the value of the annotation annotationClass from the currently executing WARP Junit class.
-   *
-   * @param annotationClass the class of the annotation to read
-   * @param testId fully qualified name of the junit test method
-   * @tparam T a subtype of Annotation
-   * @return an Option containing the annotation annotationClass from the current WARP Junit class
-   */
-  def getWarpTestClassAnnotation[T <: Annotation](annotationClass: Class[T], testId: String): Option[T] = {
-    this.getWarpTestClass(testId) flatMap { clazz => Try(Option(clazz.getAnnotation(annotationClass))).toOption.flatten }
-  }
-
-
-
-  /**
-   * Reads the value of the annotation annotationClass from the currently executing WARP Junit method.
-   *
-   * @param annotationClass the class of the annotation to read
-   * @param testId fully qualified name of the junit test method
-   * @tparam T a subtype of Annotation
-   * @return an Option containing the annotation annotationClass from the current WARP Junit method
-   */
-  def getWarpTestMethodAnnotation[T <: Annotation](annotationClass: Class[T], testId: String): Option[T] = {
-    this.getWarpTestMethod(testId) flatMap { method => Try(Option(method.getAnnotation(annotationClass))).toOption.flatten }
-  }
-
-
-
-  /**
-   * Reads the max response time from the telemetron [[Required]] annotation.
-   *
-   * @param testId fully qualified name of the junit test method
-   * @return max response time as a [[Duration]] for the test we are about to invoke
-   */
-  def getRequiredMaxValue(testId: String): Duration = {
-    this.getWarpTestMethodAnnotation(classOf[Required], testId)
-      .map(req => Duration.ofNanos(TimeUtils.toNanos(req.maxResponseTime, req.timeUnit)))
-      .getOrElse(Duration.ofMillis(-1))
-  }
-
-
-  /**
-    * Reads the max response time from the junit [[Timeout]] annotation.
+    * Reads the value of the specified annotation from the identified test.
     *
+    * @param annotationClass the class of the annotation to read
     * @param testId fully qualified name of the junit test method
-    * @return max response time as a [[Duration]] for the test we are about to invoke
+    * @tparam T a subtype of Annotation
+    * @return an Option containing the annotation annotationClass from the current WARP Junit method
     */
-  def getTimeoutValue(testId: String): Duration = {
-    this.getWarpTestMethodAnnotation(classOf[Timeout], testId)
-      .map(timeout => Duration.ofNanos(TimeUtils.toNanos(timeout.value, timeout.unit)))
-      .getOrElse(Duration.ofMillis(-1))
+  def getWarpTestAnnotation[T <: Annotation](annotationClass: Class[T], testId: TestId): Option[T] = {
+    for {
+      m <- testId.maybeTestMethod.toOption
+      a <- AnnotationUtils.findAnnotation(m, annotationClass).toOption
+    } yield a
```

Comment:
Would it be fair to throw type annotations on here? Also, could those ending `toOption`s get moved after the yield? Final minor nit, not too hot on single letter variable names, but not a big deal if it's left alone

Reply:
i can add type annotations for sure. 

i dont think i can do much nicer about the `toOption` calls. we need to work within the same monad for every line of the for-comprehension. `maybeTestMethod` is a `Try`, and `findAnnotation` returns a java `Optional`. this works but i dont think its an improvement:
```
    (for {
      m <- testId.maybeTestMethod
      a <- AnnotationUtils.findAnnotation(m, annotationClass).toTry
    } yield a).toOption
```

Reply:
Oh, okay. Yeah I prefer the way you've got it now for sure